qualification/conftest.py

```diff
@@ -248,6 +251,56 @@ def log_failure(msg="", check_str="", tb=None):
     return reporter
 
 
+@pytest.fixture(scope="session")
+def _array_compare_counter() -> Iterator[int]:
+    """Counter used to give unique filenames to array dumps."""
+    return itertools.count(0)
+
+
+def _unwrap_pytest_approx(a: np.ndarray) -> np.ndarray:
+    """Unwrap an array that has possibly been wrapped in :func:`pytest.approx`."""
+    # pytest doesn't explicitly expose this class, so we have to infer it
+    approx_cls = type(pytest.approx(np.array([1])))
+    if a.shape == () and isinstance(a[()], approx_cls):
+        return a[()].expected
+    return a
```

Comment:
I'll be honest:
1. I didn't know/forgot you could index an array with a tuple (let alone an empty tuple).
2. I'm going to trust you on the logic and syntax here. Don't quite have it in me to do a deep dive into this at the moment.

Reply:
When you do `a[i, j]`, it's actually just shorthand for `a[(i, j)]`. An empty tuple is how you get the singleton value from a zero-dimensional array.